cal/api/views.py

```diff
@@ -252,3 +252,11 @@ class TagDetailEventTimeSeries(APIView):
     def get(self, request, *args, **kw):
         tag = Tag.objects.get(user=self.request.user, id=self.kwargs['pk'])
         return Response(tag.get_time_series(self.request.query_params.get('timezone'), time_step=self.kwargs['time_step']))
+
+
+class TagsByCategories(APIView):
+
+    def get(self, request, *args, **kw):
+        categories = Category.objects.filter(user=self.request.user)
+        tag = Tag.objects.get(user=self.request.user, id=self.kwargs['pk'])
+        return Response(tag.get_category_stats(categories))
```

Comment:
Cleaner to use the related_name here: `tag.get_category_stats(self.request.user.categories)`